helix-term/src/ui/prompt.rs

```diff
@@ -32,14 +32,15 @@ pub struct Prompt {
     next_char_handler: Option<PromptCharHandler>,
 }
 
+#[repr(u8)]
 #[derive(Clone, Copy, PartialEq)]
 pub enum PromptEvent {
     /// The prompt input has been updated.
-    Update,
+    Update = 1 << 0,
     /// Validate and finalize the change.
-    Validate,
+    Validate = 1 << 1,
     /// Abort the change, reverting to the initial state.
-    Abort,
+    Abort = 1 << 2,
 }
```

Comment:
Oh, I thought you are going to use `bitflags` crate. That one is more ergonomic.

Reply:
I gave it a shot and it's a tad more complicated: If we make `PromptEvent` itself bitflags, The match arms would only exhaust after all possible combinations of the flags. The TypeableCommand functions technically only can receive one type of event at once, thus causing this code:

```rust
match event {
  PromptEvent::Abort => {},
  PromptEvent::Update => {},
  PromptEvent::Validate => {},
}
```

to error with:

```
    |
607 |     match event {
    |           ^^^^^ patterns `PromptEvent { bits: 0_u8 }`, `PromptEvent { bits: 3_u8 }` and `PromptEvent { bits: 5_u8..=u8::MAX }` not covered
    |
```

I think at least for now, it seems more correct to not allow those other possibilities, so I went with a more traditional solution